Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -181,8 +181,9 @@ bool loadCanceled
       notificationPublisher.SendStatusNotification(document.TextDocumentItem, CompilationStatus.VerificationStarted);
 
       var progressReporter = CreateVerificationProgressReporter(document);
-      document.VerificationTasks!.BatchCompletions.Subscribe(progressReporter.ReportAssertionBatchResult);
-      var implementationTasks = document.VerificationTasks!.Tasks;
+      var subscription = verifier.BatchCompletions.Subscribe(progressReporter.ReportAssertionBatchResult);
```

Comment:
I'm not sure this is correct. On the deleted code, we correctly subscribe all notifications for only the verification tasks for a given document.
On the new code, every document verification process subscribes to all messages for all batch results.

This is not caught by a test because I don't think there is a test to verify two documents with multiple splits at the same time.

The problem is that the outputPrinter is an option of the execution engine, so if we want one execution engine only, and there can be only one printer, we need to do routing ourselves.

One solution is to, in ReportAssertionBatchResult, verify that the split result is for the current document; and if not, we just ignore the message. That's one routing option.

Also, when do you unsubscribe these progressReporter?

Reply:
Nicely spotted. Pushed a new revision. The unsubscription happens on cancellation.

I spent a little bit of time today trying to refactor Boogie, splitting the options into one part that is passed to CheckerPool and cannot be changed, and another part that can be changed on each run and would take the Printer that ideally we want to change each run, but that seemed like quite a lot of work, so I discontinued and thought we could get away with something like this.